table/tables/partition.go

```diff
@@ -185,3 +185,15 @@ func (t *PartitionedTable) AddRecord(ctx sessionctx.Context, r []types.Datum, sk
 	tbl := t.GetPartition(pid)
 	return tbl.AddRecord(ctx, r, skipHandleCheck)
 }
+
+// RemoveRecord implements table.Table RemoveRecord interface.
+func (t *PartitionedTable) RemoveRecord(ctx sessionctx.Context, h int64, r []types.Datum) error {
+	partitionInfo := t.meta.GetPartitionInfo()
+	pid, err := t.locatePartition(ctx, partitionInfo, r)
```

Comment:
Delete operation will read rows first and then delete them. So I think the partition is already located when reading data. We should always call the RemoveRecord function of Partition.

Reply:
It's 
```
func (t *PartitionedTable) RemoveRecord
```
not 
```
func (t *Partition) RemoveRecord
```

This function implements **delete a record from a PartitionedTable**, it doesn't care how delete operation is implemented or needs to know anything about the caller's assumption.

Reply:
I mean how will the RemoveRecord be used in the delete executor?

Reply:
@shenli when we read the data from partition table, we ignored the partition ID in the key, so there is no easy way to use the partition ID to delete the row.